metrics/web/lib/auth/presentation/pages/loading_page.dart

```diff
@@ -70,17 +77,37 @@ class _LoadingPageState extends State<LoadingPage>
   /// Subscribes to authentication updates.
   void _subscribeToAuthUpdates() {
     _authNotifier = Provider.of<AuthNotifier>(context, listen: false);
-    _authNotifier.addListener(_authNotifierListener);
+
+    _authNotifier.addListener(_initializationListener);
+
     WidgetsBinding.instance.addPostFrameCallback((_) {
       _authNotifier.subscribeToAuthenticationUpdates();
     });
   }
 
-  /// Navigates from loading page corresponding to authentication status.
-  void _authNotifierListener() {
+  /// Subscribes to instant config updates.
+  void _subscribeToInstantConfigUpdates() {
+    _instantConfigNotifier = Provider.of<InstantConfigNotifier>(
+      context,
+      listen: false,
+    );
+
+    _instantConfigNotifier.addListener(_initializationListener);
+
+    _instantConfigNotifier.setDefaults();
+
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      _instantConfigNotifier.initializeInstantConfig();
+    });
+  }
+
+  /// Navigates to either the dashboard or the login page based on the
+  /// instant config loading and a user logged-in statuses.
+  void _initializationListener() {
```

Comment:
Let's make separate listeners for both notifiers and call a new `_navigateIfLoaded` method from every individual listener. There also could be a bool getter `_isLoaded` or `_isLoadingCompleted` that would calculate if the loading is finished or not. 

The main reason is that separate listeners are better than the one that knows everything about how to work with different notifiers. If logic with one notifier becomes more complex, the method that navigates on loading complete shouldn't know about this complexity - it should know how to perform what it is responsible for and nothing else.